### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by armen-h**
A user can now search for an item. The search can be performed in the general overview as well as in the user's listings and favorites. Uppercase and lowercase searches are possible.

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            new ActivityTestRule<>(
+                    SalesOverview.class,
+                    true,
+                    true);
+
+    @BeforeClass
+    public static void init() {
+        useMockDataStore();
+        LiteListing litelisting1 = new LiteListing("1", "listing1", "1", "Furniture");
+        litelisting1.save();
```
Comment (Walterieux) [inline app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java:46]: I think it's "safer" to Tasks.await(...).

File: app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        onView(withId(R.id.search)).perform(typeText("My title")).perform(pressKey(KeyEvent.KEYCODE_ENTER));
+        Map<String, String> map = new LinkedHashMap<>();
+        map.put("1", "listing1");
+        assertEquals(map, DataHolder.getInstance().getDataMap());
+    }
+
+    @Test
+    public void searchTriggersSearchActivity() {
+        Intents.init();
+        onView(withId(R.id.search)).perform(typeText("My title")).perform(pressKey(KeyEvent.KEYCODE_ENTER));
```
Comment (Walterieux) [inline app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java:60]: Call closeSoftKeyboard() after typeText(...).

File: app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        intended(allOf(hasComponent(SearchListings.class.getName()), hasAction(Intent.ACTION_SEARCH)));
+        Intents.release();
+    }
+
+    @Test
+    public void searchTriggersNoSearchResults() throws InterruptedException {
+        Intents.init();
+        activityRule.launchActivity(new Intent());
+        onView(withId(R.id.search)).perform(typeText("Dude")).perform(pressKey(KeyEvent.KEYCODE_ENTER));
```
Comment (Walterieux) [inline app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java:69]: closeSoftKeyboard()

File: app/src/main/res/drawable-anydpi/ic_search.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24"
+    android:tint="#FFFFFF"
+    android:alpha="0.8">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M15.5,14h-0.79l-0.28,-0.27C15.41,12.59 16,11.11 16,9.5 16,5.91 13.09,3 9.5,3S3,5.91 3,9.5 5.91,16 9.5,16c1.61,0 3.09,-0.59 4.23,-1.57l0.27,0.28v0.79l5,4.99L20.49,19l-4.99,-5zM9.5,14C7.01,14 5,11.99 5,9.5S7.01,5 9.5,5 14,7.01 14,9.5 11.99,14 9.5,14z"/>
```
Comment (Walterieux) [inline app/src/main/res/drawable-anydpi/ic_search.xml:10]: Hard coded values

File: app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java (hunk 1 above)
Comment (armen-h) [reply app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java:46]: Done

File: app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java (hunk 2 above)
Comment (armen-h) [reply app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java:60]: done

File: app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java (hunk 3 above)
Comment (armen-h) [reply app/src/androidTest/java/ch/epfl/polybazaar/search/SearchListingsTest.java:69]: done

File: app/src/main/res/drawable-anydpi/ic_search.xml (hunk 4 above)
Comment (armen-h) [reply app/src/main/res/drawable-anydpi/ic_search.xml:10]: This comes from the Android standard icon library

**Pull request merged**